### PR TITLE
Topic4

### PR DESCRIPTION
**Pull request by GBurgardt**

### CODE REVIEW

File: 04-HTML5-APIs/src/handleCanvas.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+function drawBasicFigures(){
+  drawCircle();
+  drawTriangle();
+  drawSquare();
+}
+
+function drawCircle(){
+  let c = document.getElementById('basicCircle');
```
Comment (afiorenza) [inline 04-HTML5-APIs/src/handleCanvas.js]: `c` is not a descriptive name for a variable. Please refactor this.

File: 04-HTML5-APIs/src/handleCanvas.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  let ctx = c.getContext("2d");
+  ctx.beginPath();
+  ctx.arc(75,75,60,0,2*Math.PI);
+  ctx.strokeStyle = generateColourRand();
+  ctx.stroke();
+}
+
+function drawTriangle() {
+  let c = document.getElementById('basicTriangle');
```
Comment (afiorenza) [inline 04-HTML5-APIs/src/handleCanvas.js]: Same as above

File: 04-HTML5-APIs/src/handleDragAndDrop.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+function allowDrop(ev) {
```
Comment (afiorenza) [inline 04-HTML5-APIs/src/handleDragAndDrop.js]: Remember to add more descriptive name for parameters and variables...

File: 04-HTML5-APIs/src/webSocket.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+let wsUri = "ws://echo.websocket.org/";
+let output;
+
+function testWebSocket()
+{
```
Comment (afiorenza) [inline 04-HTML5-APIs/src/webSocket.js]: `{` should go in the above line

File: 04-HTML5-APIs/src/webSocket.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  websocket = new WebSocket(wsUri);
+  websocket.onopen = function(evt) {
+    writeToScreen("CONNECTED");
+    doSend("WebSocket works");
+  };
+  websocket.onclose = function(evt) {
+    writeToScreen("DISCONNECTED");
+  };
+  websocket.onmessage = function(evt) {
+    writeToScreen('<span style="color: blue;">RESPONSE: ' + evt.data+'</span>');
```
Comment (afiorenza) [inline 04-HTML5-APIs/src/webSocket.js]: Style should go on the css file. And set a class here.

File: 04-HTML5-APIs/src/handleCanvas.js (hunk 1 above)
Comment (GBurgardt) [inline 04-HTML5-APIs/src/handleCanvas.js]: Ok!

File: 04-HTML5-APIs/src/handleCanvas.js (hunk 2 above)
Comment (GBurgardt) [inline 04-HTML5-APIs/src/handleCanvas.js]: ok!

File: 04-HTML5-APIs/src/webSocket.js (hunk 4 above)
Comment (GBurgardt) [inline 04-HTML5-APIs/src/webSocket.js]: ok!

**Pull request merged**